### PR TITLE
Resume audio context when Vision Pro enters XR immersive mode

### PR DESCRIPTION
**Pull request by docEdub**
The Apple Vision Pro suspends the audio context when entering XR immersive mode.

This change fixes the issue by listening to audio context state changes when XR immersive mode is entered, and resuming the audio context after it is suspended if the user has previously unlocked the audio engine with a user interaction.

Tested on playground https://playground.babylonjs.com/#KR5CRQ#17.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15168/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15168/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15168/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15168/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15168/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15168/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15168/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15168/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15168/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15168/merge/testResults/webgpuplaywright/index.html

**Review by RaananW — COMMENTED** [review]
Just a thought, and I will take your recommendation here - 
maybe add that as a feature in the webxr experience helper instead of here - this way if someone decides to change the state it will not be forced back to resume.

File: packages/dev/core/src/Audio/audioEngine.ts [hunk 1]
```diff
@@ -175,6 +175,23 @@ export class AudioEngine implements IAudioEngine {
         }
     }
 
+    /** @internal */
+    public _resumeAudioContextOnStateChange(): void {
+        this._audioContext?.addEventListener(
```
Comment (ryantrem) [inline packages/dev/core/src/Audio/audioEngine.ts:180]: Does `_audioContext` outlive `AudioEngine`? Is it a problem that we never remove this event listener? Will we have a leak? Also if you enter XR multiple times, will we end up registering the event handler multiple times?

Comment (docEdub) [reply packages/dev/core/src/Audio/audioEngine.ts:180]: The options object passed in prevents this. The `once: true` and 3 second abort signal make sure it's removed.

**Pull request merged**